### DISCUSSION TITLE
Clear Form After Updating Row

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#8874] [Help]
Hi,

I have a table with multiple columns including my amount column.
The table looks like this:

![image](https://user-images.githubusercontent.com/95284278/204770727-384b774c-bf2f-4aae-b163-3b80fb49fe0d.png)

I can either update it by +/- _1_ by clicking on the icons or by +/- _my preferred amount_ by entering that into the numbers field.

That works, the problem tho is clearing the field after updating the row.

I do that with the "Clear form" action which works for the last row at all times.
When I update, for Example, the second row with my preferred amount it updates the amount but it doesn't clear the form field.

Is that a bug?

**Comment by melohagan** [comment]
Hey @kasgithub8 

Could you share a screenshot of your component tree please? An app export would be useful if you can provide that.

**Reply by kasgithub8** [reply]
Hi @melohagan,

I'm so sorry for not answering.
Here's the screenshot of my component tree:
![image](https://user-images.githubusercontent.com/95284278/206660892-8fe1762f-7432-4278-a103-b57726b45883.png)

**Reply by melohagan** [reply]
Thank you @kasgithub8 

It's because you are duplicating a field that has the same identifying name multiple times within the same form. 
So when you **Clear Form** it will look for a single field via the provided name, in this case clearing only the last row.

There is a related issue here: https://github.com/Budibase/budibase/issues/5944

**Reply by kasgithub8** [reply]
@melohagan thank you for your quick answer.

So, reading through the issue, could I use a repeater here for single field saves?

**Reply by melohagan** [reply]
So what you would need to do is use the Repeater to repeat the Form and the Number field. 
Therefore each row would have a unique form and could be saved, but only with buttons within that repeater which could be a bit clunky.

**Reply by kasgithub8** [reply]
That means the form needs to be inside the table, right? 
Otherwise it repeats my table.
